### ISSUE TITLE
"Detection" is a molecular function

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-ontology#18097]
Following up on the discussion in #17977 

This is the plan for 'detection of substances' 
 - [ ] detection of oxygen (12 EXP) why not use a receptor or signalling pathway?
-> suggest obsoletion, consider: MF GO:0019826 oxygen sensor activity + BP GO:0032364 oxygen homeostasis
 - [ ]  detection of nutrient (2 EXP) ditto
-> suggest obsoletion and reannotation; papers don't seem to talk about the compounds detected as nutrients
 - [ ]  detection of cAMP (1 EXP) ditto
-> suggest obsoletion, consider: GO:0030552 cAMP binding
 - [ ]  detection of pheromone (20 EXP mainly IEP) ditto
-> suggest obsoletion, consider GO:0016503 pheromone receptor activity
 - [ ]  detection of glucose (11 annotations) ditto
-> suggest obsoletion, and create 'glucose sensor activity'

The whole branch needs to be looked at: 

* GO:0005513 detection of calcium ion
*  GO:0009730 detection of carbohydrate stimulus
* GO:0003031 detection of carbon dioxide
*  GO:0050907 detection of chemical stimulus involved in sensory perception
* GO:0010108 detection of glutamine
*  GO:0009720 detection of hormone stimulus
* GO:0003030 detection of hydrogen ion
*  GO:0002007 detection of hypoxic conditions in blood by chemoreceptor signaling
*  GO:0009754 detection of jasmonic acid stimulus
*  GO:0002236 detection of misfolded protein
* GO:0071703 detection of organic substance
* GO:0010247 detection of phosphate ion
*GO:0051776 detection of redox state
* GO:0009752 detection of salicylic acid stimulus
* GO:0002235 detection of unfolded protein

**Comment by krchristie** [comment]
It seems odd to me that `detection of stimulus` is an is_a child of `response to stimulus`. It seems to me that **detection of stimulus** has to happen BEFORE **response to stimulus** can occur. So, I think that it could be true to say that `detection of stimulus` is an is a part_of `response to stimulus`, but not that `detection of stimulus` is a type of `response to stimulus`.

**Comment by ukemi** [comment]
I think when we originally modeled it, it was a part. In a cell, it was followed by signal transduction. At a multicellular organismal level, response can be more complex.

**Comment by RLovering** [comment]
I would like detection of stimulus to remain a part of the cells response to the stimulus. in order for a cell to respond to a stimulus it has to detect it and initiate a change in the cell. But this might be a bit like what we have done with ligands being part of the signaling pathway. LIgands are the stimulus which bind the receptor which leads to activation of the signaling pathway. It does feel odd to have the ligand annotated to the signaling pathway but if the signaling pathway doesn't start with the ligand the ligand wouldn't  be associated with a bp.

**Comment by addiehl** [comment]
Yes, in #2013 and incorporated in #3447, detection of terms were modeled as part of their respective response to terms.  At a later point, it was decided that the process of detection of X was in fact a type of response to X, and the relation became is_a.  Detection as a process in this original conception included both the binding and initial signal transduction event, and so worked as either part of or is a response to.

Now it seems the conception of what detection is is changing to be the single step process of binding a substance by a receptor for that substance.  This works well for chemicals and proteins, but seems less intuitive for detection of organisms.  Also it really changes what we mean by detection to mean only the binding event.  I disagree with this change, but I really don't care, since in 5-10 years, we will just change it back.
